src/scatter.h

```diff
@@ -33,222 +35,335 @@ template <class T = float> class FormFactorSphere {
 };
 
 /**
- * @brief Unity form factor (q independent)
+ * @brief Unity form factor (q independent).
  */
 template <class T = float> struct FormFactorUnity {
     template <class Tparticle> T operator()(T, const Tparticle &) const { return 1; }
 };
 
 /**
- * @brief Calculates scattering intensity, I(q) using the Debye formula
+ * @brief Calculate scattering intensity, I(q), on a mesh using the Debye formula.
+ *
+ * It is important to note that distances should be calculated without periodicity and if molecules cross
+ * periodic boundaries, these must be made whole before performing the analysis.
  *
- * It is important to note that distances should be calculated without
- * periodicity and if molecules cross periodic boundaries, these
- * must be made whole before performing the analysis.
  * The JSON object is scanned for the following keywords:
  *
- * - `qmin` Minimum q value (1/angstrom)
- * - `qmax` Maximum q value (1/angstrom)
- * - `dr` q spacing (1/angstrom)
- * - `cutoff` Cutoff distance (angstrom). *Experimental!*
+ * - `qmin` minimum q value (1/angstrom)
+ * - `qmax` maximum q value (1/angstrom)
+ * - `dq` q mesh spacing (1/angstrom)
+ * - `cutoff` cutoff distance (angstrom); *Experimental!*
  *
- * See also <http://dx.doi.org/10.1016/S0022-2860(96)09302-7>
+ * @see http://dx.doi.org/10.1016/S0022-2860(96)09302-7
  */
-template <class Tformfactor, class Tgeometry = Geometry::Chameleon, class T = float> class DebyeFormula {
-  protected:
-    Tformfactor F;    // scattering from a single particle
-    Tgeometry geo;    // geometry to use for distance calculations
-    std::map<T, T> I; //!< Sampled, average I(q)
-    std::map<T, T> S; //!< Weighted number of samplings
-    T rc;
-
-  public:
-    T qmin, qmax, dq;
-
-    DebyeFormula(const json &j) {
-        geo = R"( { "type": "sphere", "radius": 1e9 } )"_json;
-        dq = j.at("dq").get<double>();
-        qmin = j.at("qmin").get<double>();
-        qmax = j.at("qmax").get<double>();
-        rc = j.value("cutoff", 1.0e9);
+template <class Tformfactor, class T = float> class DebyeFormula {
+    static constexpr T r_cutoff_infty = 1e9; //<! a cutoff distance in angstrom considered to be infinity
+    T q_mesh_min, q_mesh_max, q_mesh_step; //<! q_mesh parameters in inverse angstrom; used for inline lambda-functions
 
-        if (dq <= 0 || qmin <= 0 || qmax <= 0 || qmin > qmax)
-            throw std::runtime_error("DebyeFormula: invalid q parameters");
+    /**
+     * @param m mesh point index
+     * @return the scattering vector magnitude q at the mesh point m
+     */
+    #pragma omp declare simd uniform(this) linear(m:1)
+    T q_mesh(int m) {
```

Comment:
`inline`

Reply:
Done, though the function is already `inline` implicitly.
> A function defined entirely inside a class/struct/union definition, whether it's a member function or a non-member friend function, is implicitly an inline function.